palette.js

```diff
@@ -11,91 +11,82 @@ class View {
   }
 }
 
+class PaletteObject {
+  constructor(input, container, lockIcon, copyIcon, arrowIcon) {
+    this.input = input;
+    this.container = container;
+    this.lockIcon = lockIcon;
+    this.copyIcon = copyIcon;
+    this.arrowIcon = arrowIcon;
+  }
+}
+
 class PaletteController {
-  constructor(
-    paletteOneInput,
-    paletteOne,
-    paletteTwoInput,
-    paletteTwo,
-    paletteThreeInput,
-    paletteThree,
-    paletteFourInput,
-    paletteFour,
-    paletteFiveInput,
-    paletteFive,
-    submission
-  ) {
-    this.paletteOneInput = paletteOneInput;
-    this.paletteOne = paletteOne;
-    this.paletteTwoInput = paletteTwoInput;
-    this.paletteTwo = paletteTwo;
-    this.paletteThreeInput = paletteThreeInput;
-    this.paletteThree = paletteThree;
-    this.paletteFourInput = paletteFourInput;
-    this.paletteFour = paletteFour;
-    this.paletteFiveInput = paletteFiveInput;
-    this.paletteFive = paletteFive;
+  constructor(paletteObjectArr, submission) {
+    this.paletteObjectArr = paletteObjectArr;
```

Comment:
could be nice to validate palletObjectArr has valid PalletObjects?

Reply:
`if (this.paletteObjectArr) {
For each (this.paletteObjectArr as pallet)
{
      if (palle.isValid() == false)
      {
             throw new Exception("PalletObjectArray must contain PalletObjects");
      }
}

}`